### PR TITLE
Feature/29900 - Component with typography

### PR DESCRIPTION
**Pull request by kjugi**
+ Edit typography classes 
+ Add font color 
+ Add levels of fonts

https://dev.snowdog.pro/issues/29900

### CODE REVIEW

File: components/general/fonts/_fonts.scss [hunk 1]
```diff
@@ -1 +1,16 @@
-// this file probably shouldn't be empty...
+.font {
```
Comment (Igloczek) [inline components/general/fonts/_fonts.scss]: I'm not sure about the use case of this component.  IMO it should just load font files, b/c using a component to set just `font-weight` is a misuse.

PS. All components in `general` are designed to be loaded once as a base or be a source for others, this one doesn't fit that rules.

File: components/general/typography/_typography.scss [hunk 2]
```diff
@@ -1,100 +1,67 @@
...
-}
+    font-family: $font-family;
+    font-weight: 400;
+    margin: 0 0 15px;
+    font-size: 14px;
+    text-transform: uppercase;
 
-h1 {
-    font-size: 24px;
+    &--bodoni-font {
```
Comment (Igloczek) [inline components/general/typography/_typography.scss]: Nope nope nope
1. Do not define fonts as modifications, especially using the name of a font.
2. This is a base for other projects, why someone might want to use `Bodoni`?

File: components/general/typography/_typography.scss [hunk 3]
```diff
@@ -1,100 +1,67 @@
...
+        @include mq($screen-m) {
+            font-family: 'Bodoni-font' $font-family;
+        }
 
-    @include mq($screen-m) {
-        font-size: 18px;
+        &--italic {
+            font-family: 'Futura-font-italic' $font-family;
+
+            @iclude mq($screen-m) {
```
Comment (Igloczek) [inline components/general/typography/_typography.scss]: Typo.

File: components/general/typography/typography.hbs [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
+First level heading
+</h1>
+<h1 class="heading heading--second-level">
+Second level heading
+</h1>
+<h1 class="heading heading--third-level">
+Third level heading
+</h1>
+<h1 class="heading heading--fourth-level">
+Fourth level heading
```
Comment (Igloczek) [inline components/general/typography/typography.hbs]: Formatting.

File: components/general/variables/_variables.scss [hunk 5]
```diff
@@ -29,7 +28,7 @@ $screen-xxl: 1600px;
 $font-family: 'Open Sans', sans-serif;
 
 $font-size-base: 14px;
-$font-color: $color-gray-dark;
+$font-color: #4d4d4d;
```
Comment (Igloczek) [inline components/general/variables/_variables.scss]: It's perfectly fine to use predefined color.

File: components/general/typography/_typography.scss (hunk 3 above)
Comment (kjugi) [inline components/general/typography/_typography.scss]: Repaided

File: components/general/typography/typography.hbs (hunk 4 above)
Comment (kjugi) [inline components/general/typography/typography.hbs]: Fixed

File: components/general/variables/_variables.scss (hunk 5 above)
Comment (kjugi) [inline components/general/variables/_variables.scss]: Changed with another variable `$color-gray-dark`

File: components/general/fonts/_fonts.scss (hunk 1 above)
Comment (kjugi) [inline components/general/fonts/_fonts.scss]: Hmm, so I suggest to delete this styles and just add files with fonts (in this case Futura and Bodoni) but the question is how are you adding fonts in webs? Can I use fontsquirrel for that or you have your own loading font? I have all from Vasyl in files.

File: components/general/typography/_typography.scss (hunk 2 above)
Comment (kjugi) [inline components/general/typography/_typography.scss]: Yes, I agree with that, this is useful. I think to delete this classes where `font-family` exists and just left it to developers when will coding site and they will add this font which is necessary at the moment. But the question is how do you add fonts in fractal?

**Pull request merged**